### ISSUE TITLE
Why the invalid actions are allowed?

### ISSUE THREAD

**Issue by wddabc** [clab/language-universal-parser#3]
Hi, great work on this paper. I'm currently reading the code and found a part which I don't quite understand:

https://github.com/clab/language-universal-parser/blob/master/parser/lstm-parse.cc#L1193

where you allowed the invalid actions. Is this a bug or for some empirical reasons?

**Comment by miguelballesteros** [comment]
I'd say that only valid actions are explored: 
https://github.com/clab/language-universal-parser/blob/master/parser/lstm-parse.cc#L1207

this is probably just a debugging comment that we should remove.

**Comment by wddabc** [comment]
Thanks for prompt reply! You are correct. But I'd say there is still some difference. As far as I can understand, the `adiste` should be normalized over the valid actions, but the implementations here is normalized over all actions. I think it is a minor issue but the probability of the correct action will be falsely lower?

**Comment by wammar** [comment]
Somehow I was not notified of this issue until Miguel emailed me about it. I consciously computed the softmax overall all actions because I wanted the model so that the loss function penalizes high probability on invalid, incorrect actions. As the code suggests, I've experimented with both configurations but I don't recall this being an important consideration.

**Comment by wddabc** [comment]
Thanks!
